src/oci-umount.c

```diff
@@ -513,50 +634,39 @@ static int prestart(
 	}
 	memset((void *)mapped_paths, 0, (MAX_MAPS + 1) * sizeof(char *));
 
-	/* Parse oci-umounts.conf file, canonicalize path names and skip
-	 * paths which are not a mountpoint on host */
-	fp = fopen(MOUNTCONF, "r");
-	if (fp == NULL) {
-		if (errno == ENOENT) {
-			pr_pwarning("%s: Config file not found: %s", id, MOUNTCONF);
-			return 0;
-		}
-		pr_perror("%s: Failed to open config file: %s", id, MOUNTCONF);
+	/* Allocate one extra element and keep it zero for cleanup function */
+	configs = malloc((MAX_CONFIGS + 1) * sizeof(char *));
+	if (!configs) {
+		pr_perror("%s: Failed to malloc memory for configs array", id);
 		return EXIT_FAILURE;
 	}
+	memset((void *)configs, 0, (MAX_CONFIGS + 1) * sizeof(char *));
 
-	while ((read = getline(&line, &len, fp)) != -1) {
-		bool submounts_only = false;
-
-		/* Get rid of newline character at the end */
-		line[read - 1] ='\0';
-		read--;
-
-		if (iscomment(line))
-			continue;
-
-		if (nr_umounts == MAX_UMOUNTS) {
-			pr_perror("%s: Exceeded maximum number of supported unmounts is %d", id, MAX_UMOUNTS);
+	struct stat stat_buf;
+	if (stat(MOUNTCONF, &stat_buf) == 0) {
+		if (readconf(id, MOUNTCONF, mounts_on_host, &nr_umounts))
+			return -1;
+	} else {
+		if (errno != ENOENT) {
+			pr_perror("%s: Failed to read %s", id, MOUNTCONF);
```

Comment:
So we failed to do stat() on mountconf. Should we say that in error messages. 'Failed to do stat(%s): MOUNTCONF'